### PR TITLE
[Docs] Meshery Operator, MeshSync, Broker Troubleshooting Guide

### PR DESCRIPTION
**Pull request by leecalcote**
Signed-off-by: Lee Calcote <lee.calcote@layer5.io>

### CODE REVIEW

**Comment by l5io** [comment]
🚀 Preview for commit f338e704129e63e8f6ff96e491652f41b15bf761 at: https://656645064c0f360eb2303eb7--meshery-docs-preview.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit db77971db9b789afe88da7493fd261df438c9ac2 at: https://656645f66af33c107121b42c--meshery-docs-preview.netlify.app

File: docs/pages/reference/troubleshooting/meshery-operator-meshsync-troubleshooting.md [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+## Understanding the Status of Meshery Operator, MeshSync, and Meshery Broker
+
+The following table describes the various states of MeshSync and Meshery Broker and their implications.
+
+**MeshSync:**
+
+- **NOT ACTIVE:** Custom Resource not present.
+- **ENABLED:** Custom Resource present. MeshSync Controller is not connected to Broker.
+- **DEPLOYED:** Custom Resource present. MeshSync Controller present and healthy (what does healthy mean)?
```
Comment (MUzairS15) [inline docs/pages/reference/troubleshooting/meshery-operator-meshsync-troubleshooting.md:23]: healthy meaning, the status.phase of the pod is "Running".

File: docs/pages/reference/troubleshooting/meshery-operator-meshsync-troubleshooting.md [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+
+There are common issues Meshery users may face whiile operating the [Meshery Operator]({{site.baseurl}}/concepts/architecture/operator/) and its custom controllers, [MeshSync]({{site.baseurl}}/concepts/architecture/meshsync) and [Broker]({{site.baseurl}}/concepts/architecture/broker), that can be resolved by performing specific actions. This documentation aims to empower users by providing a set of troubleshooting tools and actions.
+
+## Understanding the Status of Meshery Operator, MeshSync, and Meshery Broker
+
+The following table describes the various states of MeshSync and Meshery Broker and their implications.
+
+**MeshSync:**
+
+- **NOT ACTIVE:** Custom Resource not present.
```
Comment (MUzairS15) [inline docs/pages/reference/troubleshooting/meshery-operator-meshsync-troubleshooting.md:21]:
```suggestion
```
Custom Resource are not present, it is Undeployed.

File: docs/pages/reference/troubleshooting/meshery-operator-meshsync-troubleshooting.md (hunk 1 above)
Comment (MUzairS15) [inline docs/pages/reference/troubleshooting/meshery-operator-meshsync-troubleshooting.md:23]:
```suggestion
- **DEPLOYED:** Custom Resource present. MeshSync Controller is present but the state is not RUNNING or ERRDISABLE, though
```

File: docs/pages/reference/troubleshooting/meshery-operator-meshsync-troubleshooting.md [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+- **ENABLED:** Custom Resource present. MeshSync Controller is not connected to Broker.
+- **DEPLOYED:** Custom Resource present. MeshSync Controller present and healthy (what does healthy mean)?
+- **RUNNING:** MeshSync pod present and in a running state.
+- **CONNECTED:** Deployed and connected to Broker.
+- **NOT ACTIVE:** Custom Resource not deployed or no NATS connection.
```
Comment (MUzairS15) [inline docs/pages/reference/troubleshooting/meshery-operator-meshsync-troubleshooting.md:26]:
```suggestion
```

File: docs/pages/reference/troubleshooting/meshery-operator-meshsync-troubleshooting.md [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+- **UNDEPLOYED:** Custom Resource not present.
+
+**Meshery Broker:**
+
+- **DEPLOYED:** Custom Resource deployed, External IP exposed.
```
Comment (MUzairS15) [inline docs/pages/reference/troubleshooting/meshery-operator-meshsync-troubleshooting.md:31]:
```suggestion
- **DEPLOYED:** External IP not exposed OR External IP exposed but Meshery Server is not connected as a client to Broker hence data is not being published.

```

File: docs/pages/reference/troubleshooting/meshery-operator-meshsync-troubleshooting.md [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+- **UNDEPLOYED:** Custom Resource deployed or External IP not exposed.
```
Comment (MUzairS15) [inline docs/pages/reference/troubleshooting/meshery-operator-meshsync-troubleshooting.md:32]:
```suggestion
- **UNDEPLOYED:** Custom Resource not deployed.
```

File: docs/pages/reference/troubleshooting/meshery-operator-meshsync-troubleshooting.md [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+2. Meshery is managing multiple clusters, some of which are not the cluster unto which Meshery Server is deployed.
+
+## Fault Scenarios
+
+Common failure situations that Meshery users might face are described below.
+
+1. No deployment of Meshery Operator, MeshSync, and Broker.
+    1. Probable cause: Meshery Server cannot connect to Kubernetes cluster; cluster unreachable or kubeconfig without proper permissions needed to deploy Meshery Operator; Kubernetes config initialization issues.
+1. Meshery Operator with MeshSync and Broker deployed, but Meshery Server is not receiving data from MeshSync or data the [Meshery Database]({{site.baseurl}}/concepts/architecture/database) is stale.
+    1. Probable cause: Meshery Server lost subscription to Meshery Broker; Broker server not expoerting external IP; MeshSync not connected to Broker; MeshSync not running; Meshery Database is stale.
```
Comment (MUzairS15) [inline docs/pages/reference/troubleshooting/meshery-operator-meshsync-troubleshooting.md:56]:
```suggestion
    1. Probable cause: Meshery Server lost subscription to Meshery Broker; Broker server not exposed to external IP; MeshSync not connected to Broker; MeshSync not running; Meshery Database is stale.
```

**Comment by leecalcote** [comment]
@MUzairS15 are we distinguishing between CR and Controller or are we just calling them MeshSync and Broker? Either way will work. Generally, the more specific, the better in terms of truly pinpointing where the issue lie.

**Comment by MUzairS15** [comment]
> @MUzairS15 are we distinguishing between CR and Controller or are we just calling them MeshSync and Broker? Either way will work. Generally, the more specific, the better in terms of truly pinpointing where the issue lie.

We are distinguishing CR and controller.

File: docs/pages/reference/troubleshooting/meshery-operator-meshsync-troubleshooting.md (hunk 1 above)
Comment (leecalcote) [reply docs/pages/reference/troubleshooting/meshery-operator-meshsync-troubleshooting.md:23]: @MUzairS15 please address this in the docs.

**Comment by leecalcote** [comment]
All suggestions accepted, @MUzairS15. Thank you very much.

**Comment by l5io** [comment]
🚀 Preview for commit da77ba4867982e2a012609166f76195dba4a50ab at: https://656769fdd967b93bc7619d87--meshery-docs-preview.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit 1c409119316cda9fd59ce66278c384922aaf30df at: https://65676b9ce50eab3cdd9a02ea--meshery-docs-preview.netlify.app

**Pull request merged**